openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -283,6 +283,39 @@ def test_get_cohort_preassigned_user(self):
             "User should be assigned to the right cohort"
         )
 
+    def test_get_cohort_multiple_preassignments(self):
+        """
+        When an email address is added to multiple cohorts, the last cohort assignment should be respected.
+        Then, when a user signs up for the course with that email address,
+        the user should automatically be added to that cohort and not a random cohort.
+        """
+        course = modulestore().get_course(self.toy_course_key)
```

Comment:
Nit: assuming this is the same setup as the previous test point, could create a helper method for lines 292-302 and call it in both test points.